helm-chart-sources/pulsar/templates/broker-sts/broker-sts-statefulset.yaml

```diff
@@ -283,16 +283,12 @@ spec:
         envFrom:
         - configMapRef:
             name: "{{ template "pulsar.fullname" . }}-{{ .Values.brokerSts.component }}"
-        {{- if .Values.brokerSts.ledger }}
         env:
-        - name: advertisedAddress
-          valueFrom:
-            fieldRef:
-              fieldPath: status.podIP
```

Comment:
@eolivelli - in order to make TLS hostname verification work within the k8s cluster, I removed this block that sets `advertisedAddress` to an IP address. The primary reason is that when a broker returns an owning broker for a lookup request, it uses the advertised address, and the client will verify the hostname based on the advertised address. Do you see any issues with making that change? When the broker is a stateful set, the resulting `advertisedAddress` appears to be `pulsar-broker-0.pulsar-broker.default.svc.cluster.local`. That is still a unique identifier that will allow extensions like KoP to make network calls to other brokers.

Reply:
we are good as far as advertisedAddress points to a address reachable by other brokers and by proxy